sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -1084,26 +1064,32 @@ mod checked {
         })
     }
 
-    pub(crate) fn make_object_value<'vm, 'state>(
-        vm: &'vm MoveVM,
-        session: &mut Session<'state, 'vm, LinkageView<'state>>,
+    pub(crate) fn make_object_value(
+        vm: &MoveVM,
+        linkage_view: &mut LinkageView,
+        new_packages: &[MovePackage],
         type_: MoveObjectType,
         has_public_transfer: bool,
         used_in_non_entry_move_call: bool,
         contents: &[u8],
     ) -> Result<ObjectValue, ExecutionError> {
         let contents = if type_.is_coin() {
             let Ok(coin) = Coin::from_bcs_bytes(contents) else {
-            invariant_violation!("Could not deserialize a coin")
-        };
+                invariant_violation!("Could not deserialize a coin")
+            };
             ObjectContents::Coin(coin)
         } else {
             ObjectContents::Raw(contents.to_vec())
         };
 
         let tag: StructTag = type_.into();
-        let type_ = load_type(session, &TypeTag::Struct(Box::new(tag)))
-            .map_err(|e| crate::error::convert_vm_error(e, vm, session.get_resolver()))?;
+        let type_ = load_type(
```

Comment:
would be nice if we had `load_struct_type` or something exposed, just to avoid this allocation (and in some cases a cloning). But it is fine for now :)

Reply:
man it took me 20 minutes to understand what we were talking about, finally got it - lol...
take a look and see if it makes sense